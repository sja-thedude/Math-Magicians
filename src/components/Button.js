/* eslint-disable react/prop-types, react/prefer-stateless-function */
import React, { Component } from 'react';

class Button extends Component {
  render() {
    const { bId, bName } = this.props;

    return (
      <button id={bId} type="button">{bName}</button>
    );
  }
}

export default Button;
