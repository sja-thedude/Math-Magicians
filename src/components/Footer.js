import React from 'react';

function Footer() {
  return (
    <footer>
      Math Magicians &copy; Copyright &nbsp;
      {new Date().getFullYear()}
      , audited by Syeda Juveria Afreen.
    </footer>
  );
}

export default Footer;
