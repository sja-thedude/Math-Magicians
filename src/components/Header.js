import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Header.css';

function Header() {
  return (
    <header>
      <Link to="/" className="link-logo">
        <h1>Math Magicians</h1>
      </Link>
    </header>
  );
}

export default Header;
