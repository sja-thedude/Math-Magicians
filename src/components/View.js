/* eslint-disable react/prop-types, react/prefer-stateless-function */
import React, { Component } from 'react';

class View extends Component {
  render() {
    const {
      vId, total, next, operation,
    } = this.props;

    return (
      <div id={vId}>
        {total}
        {operation}
        {next}
      </div>
    );
  }
}

export default View;
