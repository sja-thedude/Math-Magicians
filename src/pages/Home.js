import React from 'react';
import '../styles/Home.css';

function Home() {
  return (
    <section className="home-page">
      <h2>Welcome to our page!</h2>
      <p>
        An electronic calculator is typically a portable electronic device
        used to perform calculations, ranging from basic arithmetic to
        complex mathematics.
        <br />
        The first solid-state electronic calculator was created in the early
        1960s. Pocket-sized devices became available in the 1970s,
        especially after the Intel 4004, the first microprocessor, was
        developed by Intel for the Japanese calculator company Busicom. They
        later became used commonly within the petroleum industry (oil and
        gas).
        <br />
        Modern electronic calculators vary from cheap, give-away
        credit-card-sized models to sturdy desktop models with built-in
        printers. They became popular in the mid-1970s as the incorporation
        of integrated circuits reduced their size and cost. By the end of
        that decade, prices had dropped to the point where a basic
        calculator was affordable to most and they became common in schools.
        <br />
        Computer operating systems as far back as early Unix have included
        interactive calculator programs such as dc and hoc, and calculator
        functions are included in almost all personal digital assistant
        (PDA) type devices, the exceptions being a few dedicated address
        book and dictionary devices.
        <br />
        In addition to general purpose calculators, there are those designed
        for specific markets. For example, there are scientific calculators
        which include trigonometric and statistical calculations. Some
        calculators even have the ability to do computer algebra. Graphing
        calculators can be used to graph functions defined on the real line,
        or higher-dimensional Euclidean space. As of 2016, basic calculators
        cost little, but scientific and graphing models tend to cost more.
      </p>
    </section>
  );
}

export default Home;
