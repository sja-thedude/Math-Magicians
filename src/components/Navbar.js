import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';
import '../styles/Navbar.css';

function Navbar() {
  return (
    <nav>
      <Header />
      <ul className="list-nav">
        <li className="item-nav">
          <Link to="/" className="n-link">Home</Link>
        </li>
        <li className="item-nav">
          <Link to="/calculator" className="n-link">Calculator</Link>
        </li>
        <li className="item-nav">
          <Link to="/quote" className="n-link">Quote</Link>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
