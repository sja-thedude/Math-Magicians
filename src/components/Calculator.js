import React, { useState } from 'react';
import View from './View';
import Button from './Button';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(0);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = (value) => {
    const results = calculate({ total, next, operation }, value);
    setTotal(results.total);
    setNext(results.next);
    setOperation(results.operation);
  };

  return (
    <div id="calculator">
      <View vId="display" total={total} next={next} operation={operation} />
      <Button bId="clear" bName="AC" handleClick={handleClick} />
      <Button bId="del" bName="+/-" handleClick={handleClick} />
      <Button bId="del" bName="%" handleClick={handleClick} />
      <Button bId="div" bName="รท" handleClick={handleClick} />
      <Button bId="seven" bName="7" handleClick={handleClick} />
      <Button bId="eight" bName="8" handleClick={handleClick} />
      <Button bId="nine" bName="9" handleClick={handleClick} />
      <Button bId="mul" bName="x" handleClick={handleClick} />
      <Button bId="four" bName="4" handleClick={handleClick} />
      <Button bId="five" bName="5" handleClick={handleClick} />
      <Button bId="six" bName="6" handleClick={handleClick} />
      <Button bId="sub" bName="-" handleClick={handleClick} />
      <Button bId="one" bName="1" handleClick={handleClick} />
      <Button bId="two" bName="2" handleClick={handleClick} />
      <Button bId="three" bName="3" handleClick={handleClick} />
      <Button bId="add" bName="+" handleClick={handleClick} />
      <Button bId="zero" bName="0" handleClick={handleClick} />
      <Button bId="decimal" bName="." handleClick={handleClick} />
      <Button bId="equal" bName="=" handleClick={handleClick} />
    </div>
  );
};

export default Calculator;
