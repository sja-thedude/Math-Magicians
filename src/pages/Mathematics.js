/* eslint-disable linebreak-style */
import React from 'react';
import Calculator from '../components/Calculator';
import '../styles/Mathematics.css';

function Mathematics() {
  return (
    <section className="page-calculator">
      <div className="con-calculator">
        <h2>Let&apos;s do some math!</h2>
      </div>
      <Calculator />
    </section>
  );
}

export default Mathematics;
