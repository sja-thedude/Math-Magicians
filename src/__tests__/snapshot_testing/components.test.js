import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../../components/Calculator';
import Header from '../../components/Header';
import Navbar from '../../components/Navbar';
import Button from '../../components/Button';
import View from '../../components/View';
import Home from '../../pages/Home';
import Mathematics from '../../pages/Mathematics';
import Quotes from '../../pages/Quotes';
import Footer from '../../components/Footer';

describe('test screenshots for all components', () => {
  test('matches Calculator snapshot', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Header snapshot', () => {
    const tree = TestRenderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Navbar snapshot', () => {
    const tree = TestRenderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Button snapshot', () => {
    const tree = TestRenderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches View snapshot', () => {
    const tree = TestRenderer.create(<View />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = TestRenderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Mathematics snapshot', () => {
    const tree = TestRenderer.create(<Mathematics />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Quote snapshot', () => {
    const tree = TestRenderer.create(<Quotes />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Footer snapshot', () => {
    const tree = TestRenderer.create(<Footer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
