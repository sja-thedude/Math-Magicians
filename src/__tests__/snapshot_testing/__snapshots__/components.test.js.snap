// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Button snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`test screenshots for all components matches Calculator snapshot 1`] = `
<div
  id="calculator"
>
  <div
    id="display"
  >
    0
  </div>
  <button
    id="clear"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    id="del"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    id="del"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    id="div"
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    id="seven"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    id="eight"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    id="nine"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    id="mul"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    id="four"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    id="five"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    id="six"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    id="sub"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    id="one"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    id="two"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    id="three"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    id="add"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    id="zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    id="decimal"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    id="equal"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`test screenshots for all components matches Footer snapshot 1`] = `
<footer>
  Math Magicians © Copyright  
  2021
  , audited by Syeda Juveria Afreen.
</footer>
`;

exports[`test screenshots for all components matches Header snapshot 1`] = `
<header>
  <a
    className="link-logo"
    href="/"
    onClick={[Function]}
  >
    <h1>
      Math Magicians
    </h1>
  </a>
</header>
`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<section
  className="home-page"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    An electronic calculator is typically a portable electronic device used to perform calculations, ranging from basic arithmetic to complex mathematics.
    <br />
    The first solid-state electronic calculator was created in the early 1960s. Pocket-sized devices became available in the 1970s, especially after the Intel 4004, the first microprocessor, was developed by Intel for the Japanese calculator company Busicom. They later became used commonly within the petroleum industry (oil and gas).
    <br />
    Modern electronic calculators vary from cheap, give-away credit-card-sized models to sturdy desktop models with built-in printers. They became popular in the mid-1970s as the incorporation of integrated circuits reduced their size and cost. By the end of that decade, prices had dropped to the point where a basic calculator was affordable to most and they became common in schools.
    <br />
    Computer operating systems as far back as early Unix have included interactive calculator programs such as dc and hoc, and calculator functions are included in almost all personal digital assistant (PDA) type devices, the exceptions being a few dedicated address book and dictionary devices.
    <br />
    In addition to general purpose calculators, there are those designed for specific markets. For example, there are scientific calculators which include trigonometric and statistical calculations. Some calculators even have the ability to do computer algebra. Graphing calculators can be used to graph functions defined on the real line, or higher-dimensional Euclidean space. As of 2016, basic calculators cost little, but scientific and graphing models tend to cost more.
  </p>
</section>
`;

exports[`test screenshots for all components matches Mathematics snapshot 1`] = `
<section
  className="page-calculator"
>
  <div
    className="con-calculator"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    id="calculator"
  >
    <div
      id="display"
    >
      0
    </div>
    <button
      id="clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      id="del"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      id="del"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      id="div"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      id="seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      id="eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      id="nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      id="mul"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      id="four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      id="five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      id="six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      id="sub"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      id="one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      id="two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      id="three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      id="add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      id="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      id="decimal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      id="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;

exports[`test screenshots for all components matches Navbar snapshot 1`] = `
<nav>
  <header>
    <a
      className="link-logo"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Math Magicians
      </h1>
    </a>
  </header>
  <ul
    className="list-nav"
  >
    <li
      className="item-nav"
    >
      <a
        className="n-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="item-nav"
    >
      <a
        className="n-link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="item-nav"
    >
      <a
        className="n-link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`test screenshots for all components matches Quote snapshot 1`] = `
<section
  className="page-quote"
>
  <p>
    
  </p>
  <h3>
    - 
  </h3>
</section>
`;

exports[`test screenshots for all components matches View snapshot 1`] = `<div />`;
